src/core/CompUnit/Repository/Installation.pm6

```diff
@@ -243,6 +244,15 @@ sub MAIN(:$name, :$auth, :$ver, *@, *%) {
             $handle.close;
         }
 
+        # add `emulates` short-name lookups
+        for $dist.meta<emulates>.kv -> $provide-name, $emulate-name is copy {
+            my $id       = self!file-id($provide-name, $dist-id);
+            my $handle   = $dist.content($dist.meta<provides>{$provide-name});
+            my $content  = $handle.open(:bin).slurp(:close);
+            my $checksum = nqp::sha1(nqp::join("\n", nqp::split("\r\n",$content.decode('iso-8859-1'))));
```

Comment:
Can we get rid of the duplicated code here?

Reply:
The only thing I can think you mean is slurping the handle as `iso-8859-1`, but the file could be in any encoding (not necessarily the single encoding used throughout the code base to generate a checksum).